### DISCUSSION TITLE
N8N official support for workflow automation

### DISCUSSION THREAD

**Discussion by rickylabs** [medusajs/medusa#4804] [Feature Requests]
Dear Medusa Team,

I'd like to propose the integration of Medusa as an official node within the [N8N](https://n8n.io/) ecosystem. Given that N8N is open-source, it seems to be a perfect match with Medusa for enhancing workflow automation. An official node integration would undoubtedly elevate the user experience for both platforms.

However, if the team believes that Medusa might not be a fit for an official N8N node, we're open to exploring the possibility of creating a community node. We'd appreciate any support or collaboration from interested parties. We eagerly await your feedback on this proposal.

On a side note, I noticed the recent release of the Medusa UI Library. It's a fantastic addition, empowering users to craft exceptional plugin integrations within the Medusa Admin. We've already begun leveraging it and are thoroughly enjoying the developer experience.

For reference, here are some pertinent resources:

[Official (built-in) node](https://docs.n8n.io/integrations/builtin/)
[Community node](https://docs.n8n.io/integrations/community-nodes/)
[Custom node starter](https://github.com/n8n-io/n8n-nodes-starter)
Thank you for considering this request.

**Comment by nakulcodes** [comment]
Hi @rickylabs 

I also see the need for medusa node in n8n. If interested let's collaborate and start building it. Thanks!

**Comment by davidhotar** [comment]
That would be really great, especially these days!

**Comment by docloulou** [comment]
i started something with https://github.com/devlikeapro/n8n-openapi-node, the fork https://github.com/AvantGuardLLC/n8n-openapi-node seems fix some POST parameters issues but not fully fixed

**Comment by davidhotar** [comment]
We’re huge fans of n8n — we’re using it for almost everything in our workflows! One of the things that makes n8n so powerful is the incredible range of integrations it offers. For example, take WooCommerce: with just a few nodes, even non-technical team members (like our marketing folks) can build automations, sync data, and create powerful workflows without needing developers. It’s intuitive, accessible, and saves us tons of time.

That’s exactly why we believe having a Medusa integration in n8n would be absolutely amazing. It would open up the same kind of flexibility and "superpowers" for people using Medusa, allowing them to automate tasks, connect with third-party tools, and build highly customized workflows — all without writing a single line of code.

I know there are workflows in Medusa, but to be honest, it's too difficult for our marketing team.

And with the growing AI automation trends, n8n is already becoming a go-to platform for companies wanting to build smart, scalable automations. Adding Medusa to that ecosystem would make it even more attractive to merchants and developers who want a modern, open-source alternative to traditional (and often expensive) commerce platforms like Shopify.

On Wed, 25 Jun 2025 at 09:45, docloulou ***@***.***> wrote:

> i started something with https://github.com/devlikeapro/n8n-openapi-node,
> the fork https://github.com/AvantGuardLLC/n8n-openapi-node seems fix some
> POST parameters issues but not fully fixed
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/medusajs/medusa/discussions/4804#discussioncomment-13570486>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACH43PE6OCTSK2RCFMJIFZD3FJHRXAVCNFSM6AAAAAB5M4X5JCVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTGNJXGA2DQNQ>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 


/* -------------- signature -------------- */
user:{
name: "*David Hotař*",
title: "*CEO & Founder*",
mobile: "*+420 603 907 263*",
email: ***@***.*** ***@***.***>*",
website: "*devx.agency <http://www.devx.agency/>*",
location: "*Oldřichovice 923, Třinec 73961*"
}

**Comment by fxmb** [comment]
Any update on this? Would be insane to have a medusa node to fetch / update certain entities.

**Comment by kriss145** [comment]
Hello! 

https://github.com/codee-sh/medusa-plugin-automations - I'm working with something native and free, in the medusa directly.

**Comment by christopherjnelson** [comment]
Currently in beta testing before sending to n8n for approval: https://github.com/BlackSwampAI/n8n-nodes-medusa/

Would love some feedback!
